engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowInitializationActor.scala

```diff
@@ -89,30 +119,51 @@ case class WorkflowInitializationActor(workflowTag: String, backendsToInitialize
     }
   }
 
-  private def checkForDone(stateData: WorkflowInitializationActorData) = {
-    // TODO: Check these are actually the same?
-    val allActorResponses = stateData.successes ++ stateData.failures.keys
-    allActorResponses.size == backendsToInitialize.size
-  }
+  private def checkForDone(stateData: WorkflowInitializationActorData) = stateData.initializationActors.isEmpty
 
   whenUnhandled {
     case unhandledMessage =>
-      log.warning(s"Backends initializer for $workflowTag received an unhandled message: $unhandledMessage")
+      log.warning(s"received an unhandled message: $unhandledMessage")
       stay
   }
 
   onTransition {
     case _ -> state if state.terminal =>
-      log.debug(s"$workflowTag's initializer is terminal. Shutting down.")
+      log.debug("State is now terminal. Shutting down.")
       context.stop(self)
     case fromState -> toState =>
       // Only log this at debug - these states are never seen or used outside of the CallActor itself.
-      log.debug(s"$workflowTag's initializer is transitioning from $fromState to $toState.")
+      log.debug(s"State is transitioning from $fromState to $toState.")
   }
 
   /**
     * Based on backendAssignments, creates a BackendWorkflowInitializationActor per backend and maps the actor to the
     * name of the backend which it is initializing.
     */
-  def backendWorkflowInitializationActors:  Map[ActorRef, String] = ??? //TODO: Implement this!
+  def backendWorkflowInitializationActors(backendAssignments: Map[Call, String]):  Map[ActorRef, String] = {
+    val callAssignmentMap = callAssignments(backendAssignments)
+    val backendsToInitialize = backendAssignments.values.toSet
+    backendsToInitialize
+      .map { backend => (initializationActor(backend, callAssignmentMap(backend)), backend) } // Create the initializationActors
+      .collect { case (Some(actorRef), backend) => actorRef -> backend } // Only track the backends which need initialization
+      .toMap
+  }
+
+  /**
+    * Turns a mapping from Call => backendName into a mapping from backendName => List[Call]
+    */
+  private def callAssignments(backendAssignments: Map[Call, String]): Map[String, Seq[Call]] = {
```

Comment:
Is this the same as the code I commended on earlier? If so can it be extracted out?